### PR TITLE
Added retry for TransportException in azure blob storage

### PR DESCRIPTION
**Pull request by SmitaRKulkarni**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Added retry for TransportException during azure blob storage read and download.
Fixes issue like https://s3.amazonaws.com/clickhouse-test-reports/52835/aec0667f16ad39126306ef7133a8af5722ed687e/integration_tests__asan__%5B5_6%5D.html

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 473fb60e6fb4e27aa712435482bf49109ae3d64f with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53128/473fb60e6fb4e27aa712435482bf49109ae3d64f/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Disks/IO/ReadBufferFromAzureBlobStorage.cpp [hunk 1]
```diff
@@ -102,6 +102,19 @@ bool ReadBufferFromAzureBlobStorage::nextImpl()
     size_t bytes_read = 0;
 
     size_t sleep_time_with_backoff_milliseconds = 100;
+
+    auto handle_exception = [&, this](const auto & e, size_t i)
+    {
+        LOG_INFO(log, "Exception caught during Azure Read for file {} at attempt {}: {}", path, i, e.Message);
```
Comment (kssenii) [inline src/Disks/IO/ReadBufferFromAzureBlobStorage.cpp:108]:
```suggestion
        LOG_INFO(log, "Exception caught during Azure Read for file {} at attempt {}/{}: {}", path, i + 1, max_single_read_tries, e.Message);
```

File: src/Disks/IO/ReadBufferFromAzureBlobStorage.cpp [hunk 2]
```diff
@@ -211,6 +221,17 @@ void ReadBufferFromAzureBlobStorage::initialize()
         blob_client = std::make_unique<Azure::Storage::Blobs::BlobClient>(blob_container_client->GetBlobClient(path));
 
     size_t sleep_time_with_backoff_milliseconds = 100;
+
+    auto handle_exception = [&, this](const auto & e, size_t i)
+    {
+        LOG_INFO(log, "Exception caught during Azure Download for file {} at offset {} at attempt {} : {}", path, offset, i + 1, e.Message);
```
Comment (kssenii) [inline src/Disks/IO/ReadBufferFromAzureBlobStorage.cpp:227]:
```suggestion
        LOG_INFO(log, "Exception caught during Azure Download for file {} at offset {} at attempt {}/{}: {}", path, offset, i + 1, max_single_download_tries, e.Message);
```

**Review by kssenii — APPROVED** [review]

File: src/Disks/IO/ReadBufferFromAzureBlobStorage.cpp (hunk 2 above)
Comment (SmitaRKulkarni) [reply src/Disks/IO/ReadBufferFromAzureBlobStorage.cpp]: addressed it

**Pull request merged**